spikeextractors/extractors/neuroscopesortingextractor/neuroscopesortingextractor.py

```diff
@@ -26,17 +37,31 @@ class NeuroscopeSortingExtractor(SortingExtractor):
     is_writable = True
     mode = 'custom'
 
-    def __init__(self, resfile, clufile):
+    def __init__(self, resfile, clufile, keep_mua_units=True):
         SortingExtractor.__init__(self)
         res = np.loadtxt(resfile, dtype=np.int64, usecols=0, ndmin=1)
         clu = np.loadtxt(clufile, dtype=np.int64, usecols=0, ndmin=1)
         if len(res) > 0:
+            # Extract the number of clusters read as the first line of the clufile
+            # then remove it from the clu list
             n_clu = clu[0]
             clu = np.delete(clu, 0)
+            
+            # Initialize spiketrains and extract times from .res and appropriate clusters from .clu
+            # based on user input for ignoring multi-unit activity
             self._spiketrains = []
-            self._unit_ids = list(x + 1 for x in range(n_clu))
-            for s_id in self._unit_ids:
-                self._spiketrains.append(res[(clu == s_id).nonzero()])
+            if keep_mua_units: # default
+                n_clu -= 1;
+                self._unit_ids = list(x+1 for x in range(n_clu)) # generates list from 1,...,clu[0]-1
+                for s_id in self._unit_ids:
+                    self._spiketrains.append(res[(clu == s_id).nonzero()])
+            else:
+                # Ignoring IDs of 0 until get_unsorted_spike_train is implemented into base
+                # Also ignoring IDs of 1 since user called keep_mua_units=False
+                n_clu -= 2;
+                self._unit_ids = list(x+1 for x in range(n_clu)) # generates list from 1,...,clu[0]-2
+                for s_id in self._unit_ids:
+                    self._spiketrains.append(res[(clu == s_id+1).nonzero()]) # only reading cluster IDs 2,...,clu[0]-1
```

Comment:
Great!

Could you also add the [`get_unsorted_spike_train`](https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/sortingextractor.py#L703) method to return cluster 0?

Reply:
So, we don't need the unsorted spikes, consistent with the original implementation. Adding it into the base for SortingExtractor seems more complicated than the simple modification I made for the Neuroscope class initializer.

Unless you're thinking it's as simple as setting it to 'pass' in the base, and then adding a specific method to the Neuroscope class that keeps and returns the 0 index spikes? But would other classes know how to handle it at that point?

Reply:
There is already a base method, so it could be just overridden. However I agree with you that it might not be relevant for downstream analysis. So let's keep it this way